README.MD

```diff
@@ -0,0 +1,114 @@
+
+<div style="display:flex; gap:10px; flex-wrap:wrap">
+  <a href="https://www.npmjs.com/package/@password-validator/core" style="text-decoration:none">
+    <img src="https://img.shields.io/npm/v/@password-validator/core.svg"  />
+  </a>
+
+  <a href="https://www.npmjs.com/package/@password-validator/core" style="text-decoration:none">
+    <img src="https://img.shields.io/npm/dt/@password-validator/core.svg" />
+  </a>
+
+  <a href="https://github.com/Theshedman/password-validator.js" style="text-decoration:none">
+    <img src="https://img.shields.io/github/workflow/status/Theshedman/password-validator/build-ci?logo=github" />
+  </a>
+
+  <a href="https://github.com/Theshedman/password-validator.js" style="text-decoration:none">
+    <img src="https://img.shields.io/github/languages/top/Theshedman/password-validator.js.svg"  />
+  </a>
+</div>
+
+---
+
+
+A zero dependency library for password validation that lets you compose and combine multiple rules to create a validator that suits your needs.
+
+### Installation
+
+```bash
+npm install @password-validator/core
+```
+
+### Basic Usage
+
+```ts
+import { LowerCaseValidator, MaxLengthValidator, MinLengthValidator, PasswordValidatorManager, SpecialCharacterValidator, UpperCaseValidator } from '@password-validator/core';
+
+const pm = new PasswordValidatorManager(); // Create a password validator manager
+
+const minLength = new MinLengthValidator(8); // Minimum length of 8 characters
+const maxLength = new MaxLengthValidator(16); // Maximum length of 16 characters
+const uppercases = new UpperCaseValidator(2); // At least 2 uppercase characters
+const lowercases = new LowerCaseValidator(2); // At least 2 lowercase characters
+const specialCharacters = new SpecialCharacterValidator(2); // At least 2 special characters
+
+pm.register(minLength, maxLength, uppercases, lowercases, specialCharacters);  // Register all the validators with the manager
+
+// Use the manager to validate passwords
+const result = pm.validate('MyPassword123!*') // { valid: true, messages: [] } --> Password is valid
+const result = pm.validate('MyPassword123'); // { valid: false, messages: ['must contain at least 2 special characters.'] } --> Password is invalid
+
+```
+
+### Validators
+
+A validator is class that can validate a password and return a result. A validator can be used on its own (as shown below) or in conjunction with a `PasswordValidatorManager` (as show above) to create a more complex validator.
+
+```ts
+import { MinLengthValidator, DigitValidator } from '@password-validator/core';
+
+// MinLengthValidator
+const minLength = new MinLengthValidator(10); // Minimum length of 8 characters
+
+// Use the min length validator to validate passwords
+const result = minLength.validate('MyPassword') // { valid: true, messages: [] } --> Password is valid
+const result = minLength.validate('Password') // { valid: false, messages: ['must not be less than 10 minimum characters.'] } --> Password is valid
+
```

Comment:
It is an anti-pattern to use the validators independently on their own. We need to remove it from the docs. You can explain about the validators but not how to use them separately.

Reply:
my thoughts are that users might want to use each validator individually and they should be allowed to do so.

We'll be removing it from the docs, however, we might have to start thinking about allowing that pattern soon :)

Reply:
Password validation involves checking for more than one policy for the password to pass. To validate a password, you must check for multiple combinations of password policies like MinLength, MaxLength, etc. So, it will not be a good practice to do the validations separately (and manually).

Thus, I don't think there would be a need to use the validators directly.

Reply:
Gotcha! I was actually thinking of broader use-cases. Thanks for keeping me in scope hahaha :)